tools/ARIAtools/extractProduct.py

```diff
@@ -1019,14 +1077,33 @@ def export_products(
     mask = None if maskfile is None else osgeo.gdal.Open(maskfile)
     dem = None if demfile is None else osgeo.gdal.Open(demfile)
     dem_expanded = (
-        None if demfile_expanded is None else osgeo.gdal.Open(demfile_expanded))
+        None if demfile_expanded is None
+        else osgeo.gdal.Open(demfile_expanded))
 
     # create dictionary of all inputs needed for correction lyr extraction
+    # Get the authority code (EPSG code)
+    srs = osgeo.osr.SpatialReference()
+    srs.ImportFromWkt(proj)
+    srs.AutoIdentifyEPSG()
+    epsg_code = srs.GetAuthorityCode(None)
+    srs = None
     lyr_input_dict = {
-        'layers': layers, 'prods_TOTbbox': prods_TOTbbox, 'dem': dem_expanded,
-        'lat': lat, 'lon': lon, 'mask': mask, 'verbose': verbose,
-        'multilooking': multilooking, 'rankedResampling': rankedResampling,
-        'num_threads': num_threads}
+        'layers': layers, 'prods_TOTbbox': prods_TOTbbox,
+        'proj': int(epsg_code), 'dem': dem_expanded, 'lat': lat, 'lon': lon,
+        'mask': mask, 'verbose': verbose, 'multilooking': multilooking,
+        'rankedResampling': rankedResampling, 'num_threads': num_threads}
+
+    # track if product stack is NISAR GUNW or not
+    nisar_file = False
+    range_correction = True
+    track_fileext = full_product_dict[0]['unwrappedPhase'][0]
+    if len(track_fileext.split('.h5')) > 1:
```

Comment:
A different check will be more obvious to any reader of the code (i.e. .endswith() versus .split), it more clear == less cognitive load

Reply:
This was one of the items I cleared up in the `dev_refactor_mskupdate` branch.. So I'll leave this as is here